src/librustc_back/target/mod.rs

```diff
@@ -698,6 +707,10 @@ impl ToJson for Target {
         target_option_val!(max_atomic_width);
         target_option_val!(panic_strategy);
 
+        if self.options.max_atomic_width.to_string() != self.target_pointer_width {
```

Comment:
Cool, though if this is how it already works, I have no problem with a followup PR.